docs/v1-proposed-api.md

```diff
@@ -0,0 +1,248 @@
+  controlPlaneConfig:
+    servingCertificates:
+      default: primary-cert-bundle
+  imageSet:
+    name: openshift-v4.1.18
+  images: {}
+  ingress:
+  - domain: apps.ci-cluster-v4-1.r4f1.s1.devshift.org
+    name: default
+    servingCertificate: primary-cert-bundle
+  installed: true
+  manageDNS: true
+  networking:
+    clusterNetworks:
+    - cidr: 10.128.0.0/14
+      hostSubnetLength: 23
+    machineCIDR: 10.0.0.0/16
+    serviceCIDR: 172.30.0.0/16
+    type: OpenShiftSDN
+  platform:
+    aws:
+      region: us-east-1
+  platformSecrets:
+    aws:
+      credentials:
+        name: aws
+  pullSecret:
+    name: pull
+  sshKey:
+    name: ssh
+status:
+  adminKubeconfigSecret:
+    name: ci-cluster-v4-1-0-ndv7l-admin-kubeconfig
+  adminPasswordSecret:
+    name: ci-cluster-v4-1-0-ndv7l-admin-password
+  apiURL: https://api.ci-cluster-v4-1.r4f1.s1.devshift.org:6443
+  certificateBundles:
+  - generated: true
+    name: primary-cert-bundle
+  cliImage: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:528f2ead3d1605bdf818579976d97df5dd86df0a2a5d80df9aa8209c82333a86
+  clusterID: 1e4f00a3-a30e-46d7-9567-4bb61caf6d73
+  clusterVersionStatus:
+    [SNIP]
+  conditions:
+  - lastProbeTime: "2019-10-23T08:40:24Z"
+    lastTransitionTime: "2019-10-23T08:40:24Z"
+    message: Control plane certificates are present
+    reason: ControlPlaneCertificatesFound
+  installed: true
+  installedTimestamp: "2019-10-23T08:38:03Z"
+  installerImage: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:c9c1e6d1897b74d366d331f5cdc3fa16c5a96abe28ff8238833c09042156e983
+  provision:
+    name: ci-cluster-v4-1-0-ndv7l
+  webConsoleURL: https://console-openshift-console.apps.ci-cluster-v4-1.r4f1.s1.devshift.org
+```
+
+### Proposed Changes
+
+
+ 1. Require an InstallConfig secret, referenced in a new ClusterDeployment.Spec field, to pass directly to installer.
+   * This will allow immediate use of new openshift-install features without needing to mirror in Hive's API.
+   * Remaining items in ClusterDeployment.Spec should be assumed mutable to every extent possible.
+   * Referenced as Spec.InstallConfigSecret. Secrets used as this config file contains ssh keys and pull secrets.
+   * Hive will parse the InstallConfig and copy relevant information back to the ClusterDeployment.
+     * What should happen if a field we support copyback for is set in the ClusterDeployment as well?
+ 1. Remove:
+   * Spec.ClusterName - In install config, not relevant thereafter.
+   * Spec.ControlPlane - Not relevant after install today as masters are not a MachineSet. Can be re-added in future if necessary.
+   * Spec.Networking
+ 1. Remaining immutable ClusterDeployment.Spec fields:
+   * Spec.ImageSet and Spec.Images. (technically could be mutable but has no effect once we've installed)
+   * Spec.Platform.AWS.Region (deemed important enough to be present in spec)
+     * Should this move to status instead? Introspected from the InstallConfig? What if it were lost?
```

Comment:
`Platform.AWS.Region` is not a natural candidate for status. It is more of a immutable information for the user.

Reply:
Can it be part of `ClusterMetadata`?